x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -212,20 +205,16 @@ export const createMetricThresholdExecutor = (libs: InfraBackendLibs) =>
             .filter((result) => result[group].isNoData)
             .map((result) => buildNoDataAlertReason({ ...result[group], group }))
             .join('\n');
-        } else if (nextState === AlertStates.ERROR) {
-          reason = alertResults
-            .filter((result) => result[group].isError)
-            .map((result) => buildErrorAlertReason(result[group].metric))
-            .join('\n');
         }
       }
 
       if (reason) {
-        const firstResult = first(alertResults);
-        const timestamp = (firstResult && firstResult[group].timestamp) ?? moment().toISOString();
+        const timestamp = startedAt.toISOString();
         const actionGroupId =
           nextState === AlertStates.OK
             ? RecoveredActionGroup.id
+            : nextState === AlertStates.NO_DATA
```

Comment:
@simianhacker AlertsState refers actually to RulesState, right? The naming confuses me. Probably refactoring alerts to rules is out of scope of this PR. Shall I create another ticket to refactor wrong uses of alerts to rules?

Reply:
Yes, we should create a new ticket for standardizing variable names